src/net/socket.rs

```diff
@@ -212,6 +214,32 @@ impl Socket {
             true
         }
     }
+
+    #[cfg(test)]
+    pub fn connection_count(&self) -> usize {
```

Comment:
I prefer keeping tests inside the test module and test the API from there.

Reply:
It's not a test, just a function used only in tests.

Reply:
so why not move that function to the test module? You can call into private modules there as well.

Reply:
Because it requires `&self` (`Socket`). I've changed it now so it's private without the cfg.